Moonshot/player/PlayerArsenal.gd

```diff
@@ -5,30 +5,210 @@ class LaserBlaster:
 	var bullet = load("res://Combat/bullet.tscn")
 	var sfx = load("res://player/assets/sounds/laser_pew.ogg") 
 	var fire_speed = 0.25
+	var f_mag = 800
+	var damage = 0.4
+	var no_shots = 1
+	var shot_rotation_modifiers = [0]
 	
 	func _init():
 		sfx.set_loop(false)
 		sound.stream = sfx
 
+
+	func shoot():
+		sound.play()
+		var shots = []
+		var shot
+		for i in range(no_shots):
```

Comment:
rather than a loop of 1:

```gdscript
func shoot() -> void:
   sound.play()
   var shot = bullet.instance()
   shot.f_mag = f_mag
   shot.damage = damage
   return [shot]
```

Reply:
Fair enough, I just copied the code from the shot gun to keep it generic